storage/shard.go

```diff
@@ -726,6 +732,48 @@ func (s *dbShard) Repair(
 	namespace ts.ID,
 	tr xtime.Range,
 	repairer databaseShardRepairer,
-) (repair.MetadataComparisonResult, error) {
+) (repair.Result, error) {
 	return repairer.Repair(ctx, namespace, tr, s)
 }
+
+func (s *dbShard) UpdateSeries(
+	id ts.ID,
+	blk block.DatabaseBlock,
+	markDirty bool,
+) error {
+
+	// ensure shard is bootstrapped
+	s.Lock()
+	if s.bs != bootstrapped {
+		s.Unlock()
+		return errShardNotBootstrapped
+	}
+	s.Unlock()
+
+	// Prepare series to be updated
+	entry, err := s.writableSeries(id)
+	if err != nil {
+		return err
+	}
+
+	// Perform update
+	err = entry.series.Update(blk)
+	entry.decrementWriterCount()
+	if err != nil {
+		return err
+	}
+
+	if markDirty {
```

Comment:
I've got a TODO in the code that is going to need this - more info (here)[https://github.com/m3db/m3db/pull/214/files?diff=unified#diff-97d06de2227f251da9f03d6e77a03927R340]